core/iwasm/common/wasm_runtime_common.h

```diff
@@ -303,7 +303,6 @@ typedef struct WASMModuleCommon {
        Wasm_Module_AoT, and this structure should be treated as
        AOTModule structure. */
     uint32 module_type;
-    uint8 module_data[1];
```

Comment:
Adding `module_data[1]` here is to indicate that there is extra data in this structure, which makes the code easier to understand. My personal opinion is that we had better keep it, or add comments for it.
Is there any compilation error or execution error?